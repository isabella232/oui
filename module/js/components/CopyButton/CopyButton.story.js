import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, select } from '@storybook/addon-knobs';
import CopyButton from '../CopyButton';
var code = "const CopyButton = ({ testSection, onClick, text }) => {\n  return (\n    <CopyToClipboard\n      text={ textToCopy }\n      onCopy={ onClick }>\n      <Button\n        style=\"plain\"\n        ariaLabel=\"Copy code snippet\"\n        testSection={ testSection }>\n        <Icon name='clipboard' />\n      </Button>\n    </CopyToClipboard>\n  );\n}";
var stories = storiesOf('CopyButton', module);
stories.addDecorator(withKnobs).addDecorator(function (story) {
  return React.createElement("div", {
    id: "root-preview"
  }, story());
});
stories.add('Default', function () {
  return React.createElement(CopyButton, {
    textToCopy: code
  });
}).add('With style specified', function () {
  return React.createElement(CopyButton, {
    textToCopy: code,
    style: select('style', {
      'none': 'none (use default button style)',
      'highlight': 'highlight',
      'danger': 'danger',
      'danger-outline': 'danger-outline',
      'outline': 'outline',
      'outline-reverse': 'outline-reverse',
      'plain': 'plain',
      'toggle': 'toggle',
      'underline': 'underline',
      'unstyled': 'unstyled'
    }, 'none')
  });
});