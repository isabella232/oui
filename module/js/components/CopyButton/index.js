import React from 'react';
import PropTypes from 'prop-types';
import { CopyToClipboard } from 'react-copy-to-clipboard';
import Button from '../Button';
import Icon from 'react-oui-icons';
/**
 * @param {Object} props - Properties passed to component
 * @returns {ReactElement}
 * @private
 */

var CopyButton = function CopyButton(_ref) {
  var style = _ref.style,
      testSection = _ref.testSection,
      textToCopy = _ref.textToCopy;
  return React.createElement(CopyToClipboard, {
    text: textToCopy
  }, React.createElement(Button, {
    style: style === 'none' ? null : style,
    ariaLabel: "Copy code snippet",
    testSection: testSection ? "".concat(testSection, "-copy-button") : null
  }, React.createElement(Icon, {
    name: "clipboard"
  })));
};

CopyButton.propTypes = {
  /** Style option for the button */
  style: PropTypes.oneOf(['highlight', 'danger', 'danger-outline', 'outline', 'outline-reverse', 'plain', 'toggle', 'underline', 'unstyled', 'none']),

  /** Hook for automated JavaScript tests */
  testSection: PropTypes.string,

  /** The text or code that will be copied */
  textToCopy: PropTypes.string.isRequired
};
CopyButton.defaultProps = {
  style: 'plain'
};
export default CopyButton;