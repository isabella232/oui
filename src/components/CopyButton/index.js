import React from 'react';
import PropTypes from 'prop-types';
import { CopyToClipboard } from 'react-copy-to-clipboard';

import Button from '../Button';
import Icon from 'react-oui-icons';

/**
 * @param {Object} props - Properties passed to component
 * @returns {ReactElement}
 * @private
 */

const CopyButton = ({
  style,
  testSection,
  textToCopy,
  usesTextLabel,
  textLabel,
  onCopy,
}) => {

  return (
    <CopyToClipboard
      text={ textToCopy }
      onCopy={ onCopy }>
      <Button
        style={ style === 'none' ? null : style }
        size={ usesTextLabel ? 'tiny' : null }
        ariaLabel="Copy code snippet"
        testSection={ testSection ? `${testSection}-copy-button` : null }>
        { usesTextLabel ? textLabel : <Icon name='clipboard' /> }
      </Button>
    </CopyToClipboard>
  );
};

CopyButton.propTypes = {
  /** Function to perform when content is copied.
      This callback is passed as onCopy prop to CopyToClipboard component.
      onCopy(text, result)
      result (bool): Returns true if copied successfully, else false.
  */
  onCopy: PropTypes.func,
  /** Style option for the button */
  style: PropTypes.oneOf([
    'highlight',
    'danger',
    'danger-outline',
    'outline',
    'outline-reverse',
    'plain',
    'toggle',
    'underline',
    'unstyled',
    'none',
  ]),
  /** Hook for automated JavaScript tests */
  testSection: PropTypes.string,
  /** String label to use when usesTextLabel is true */
  textLabel: PropTypes.string,
  /** The text or code that will be copied */
  textToCopy: PropTypes.string.isRequired,
  /**
   * Whether or not this should use a string label
   * instead of an icon
   */
  usesTextLabel: PropTypes.bool,
};

CopyButton.defaultProps = {
  style: 'plain',
  usesTextLabel: false,
  textLabel: 'Copy',
};

export default CopyButton;
