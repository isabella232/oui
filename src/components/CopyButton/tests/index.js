import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import CopyButton from '../index';

jest.mock('copy-to-clipboard', () => {
  return jest.fn();
});

describe('components/CopyButton', () => {
  it('should render children the icon button', () => {
    const component = shallow(
      <CopyButton textToCopy={ 'let bingo = "BINGO"' } testSection="fake-test-section"/>
    );
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('should render default text when usesTextLabel is true', () => {
    const component = shallow(
      <CopyButton textToCopy={ 'let bingo = "BINGO"' } testSection="fake-test-section" usesTextLabel={ true } style="highlight"/>
    );
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('should render provided text when usesTextLabel is true and textLabel is given', () => {
    const component = shallow(
      <CopyButton
        textToCopy={ 'let bingo = "BINGO"' } testSection="fake-test-section"
        usesTextLabel={ true } textLabel={ 'custom-copy-label' } style="highlight"
      />
    );
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('should call onCopy callback when CopyButton is clicked', () => {
    const mockFunction = jest.fn();

    const component = mount(
      <CopyButton
        textToCopy={ 'let bingo = "BINGO"' } usesTextLabel={ true }
        testSection="fake-test-section" onCopy={ mockFunction } style="highlight"
      />
    );

    const button = component.find('CopyToClipboard');
    button.simulate('click');
    expect(mockFunction).toHaveBeenCalled();
  });

});

