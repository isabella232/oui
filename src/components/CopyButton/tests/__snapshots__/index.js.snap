// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/CopyButton should render children the icon button 1`] = `
<CopyToClipboard
  text="let bingo = \\"BINGO\\""
>
  <Button
    ariaLabel="Copy code snippet"
    isDisabled={false}
    isLink={false}
    isLoading={false}
    isSubmit={false}
    loadingText=""
    onBlur={[Function]}
    onClick={[Function]}
    onMouseDown={[Function]}
    size={null}
    style="plain"
    testSection="fake-test-section-copy-button"
    width="default"
  >
    <Memo(Icon)
      name="clipboard"
    />
  </Button>
</CopyToClipboard>
`;

exports[`components/CopyButton should render text when usesTextLabel is true 1`] = `
<CopyToClipboard
  text="let bingo = \\"BINGO\\""
>
  <Button
    ariaLabel="Copy code snippet"
    isDisabled={false}
    isLink={false}
    isLoading={false}
    isSubmit={false}
    loadingText=""
    onBlur={[Function]}
    onClick={[Function]}
    onMouseDown={[Function]}
    size="tiny"
    style="highlight"
    testSection="fake-test-section-copy-button"
    width="default"
  >
    Copy
  </Button>
</CopyToClipboard>
`;
